src/Dialog/withResponsiveFullScreen.js

```diff
@@ -0,0 +1,42 @@
+// @flow
+
+import createEagerFactory from 'recompose/createEagerFactory';
+import wrapDisplayName from 'recompose/wrapDisplayName';
+import withWidth, { isWidthDown } from '../utils/withWidth';
+import Dialog from './Dialog';
+import type { Breakpoint } from '../styles/breakpoints';
+
+type Options = { breakpoint: Breakpoint }
+
+/**
+ * Dialog will responsively be full screen _at or below_ the given breakpoint
+ * (defaults to 'sm' for mobile devices).
+ */
+function withResponsiveFullScreen(options: Options = { breakpoint: 'sm' }) {
```

Comment:
Still, I think that we can merge that HOC as already providing value and elegant 😄 .